create-quasar/utils/index.js

```diff
@@ -218,6 +218,26 @@ module.exports.ensureOutsideProject = function () {
   }
 }
 
+const defaultProjectFolder = 'quasar-project'
+
+module.exports.promptsDefaults = {
+  projectType: 'app',
+  projectFolder: defaultProjectFolder,
+  overwrite: 'n',
+  packageManager: module.exports.runningPackageManager ?? 'yarn',
+  quasarVersion: 'v2',
+  scriptType: 'js',
+  engine: 'webpack',
+  productName: 'Quasar App',
+  name: module.exports.inferPackageName(defaultProjectFolder),
+  description: 'A Quasar Project',
+  author: getGitUser(),
+  typescriptConfig: 'composition',
+  css: 'scss',
+  preset: ['lint'],
+  lintConfig: 'prettier',
+}
```

Comment:
`defaults` might become more useful for all kinds of people if it doesn't choose every option possible. It could set the following:
```suggestion
module.exports.promptsDefaults = {
  projectType: 'app',
  overwrite: 'n',
  packageManager: module.exports.runningPackageManager ?? 'yarn',
  quasarVersion: 'v2',
  productName: 'Quasar App',
  name: module.exports.inferPackageName(defaultProjectFolder),
  description: 'A Quasar Project',
  author: getGitUser(),
}
```
The most common use case is to create a Quasar v2 app, enter a folder name, then the name, author, etc. is all the defaults. The rest is just selecting between Vite and Webpack, CSS flavor, features, etc. which has lots of combinations.

If we merge this as is, lots of people may not find it useful as it's not even possible to enter a custom folder name.

Reply:
We could also try to infer the product name by converting the folder name to the capitalized case, e.g. `quasar-app` -> `Quasar App` to make it more useful than using `Quasar App`

Reply:
The point of `--defaults` flag is to totally avoid user interaction, useful for people trying to automate Quasar projects creation, it's not meant to speed up people creating projects, we would need other flags for that
The name itself explains it: it accepts all the defaults values
If you prefer, we can call it `-y` as it is in many other CLI tools, I used `defaults` because it should be easier to understand

Applying these changes would leave some questions open for interaction and that would defy the primary purpose of the flag

I'm more on the side of adding the `defaults` flag and then provide some other flag to customize the choices
Maybe it's better to define those other flags before merging this at this point

---

A suggestion to solve the folder name is to take the first argument from the command line for that, many other CLIs do that
So it would become `yarn create quasar my-project-folder --defaults`

I agree that the product name should be inferred by the folder name

Reply:
It seems this is solving an artificial "problem" of the ability to create projects without any interaction. This is good at solving that problem, but I just don't think it's practical and doubt that anybody who wanted the presets option will be happy with this option. But, if it's gonna be implemented like this I think `-y` would be better for this.

Accepting the project name as the first argument is a good idea in any case. So, we should definitely do that. I think most, if not all, scaffolding tools use that approach.

About adding more flags, as you said this PR is not really useful unless there are more flags. But, my preferred way would be to implement the defaults approach I mentioned, which will save some time for most users and is practical. Then, we could add more flags to answer more questions. If someone wants, they could provide all the flags to create a project without any interaction, in the exact way they want. This allows room for flexibility, while accepting all then overriding the rest is more limiting and would require the user to provide almost every flag possible to be practical, IMO.

Reply:
> It seems this is solving an artificial "problem" of the ability to create projects without any interaction

This may be true, we should probably seek some validation from the community before merging this in